src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -320,208 +263,194 @@ export default class QueryEditorUI extends Component<Props, State> {
-          </EuiText>,
-          <QueryResult queryStatus={this.props.queryStatus!} />,
-        ],
-        end: [
-          <EuiButtonEmpty
-            iconSide="left"
-            iconType="clock"
-            iconGap="s"
-            size="xs"
-            onClick={this.toggleRecentQueries}
-            className="queryEditor__footerItem"
-            flush="both"
-          >
-            <EuiText size="xs" color="subdued">
-              {'Recent queries'}
-            </EuiText>
-          </EuiButtonEmpty>,
-        ],
-      },
-    };
+      // Optional: Cleanup on component unmount
+      return () => {
+        disposable.dispose();
+      };
+    },
+    provideCompletionItems,
+    prepend: props.prepend,
+    footerItems: {
+      start: [
+        <EuiText size="xs" color="subdued" className="queryEditor__footerItem">
+          {`${lineCount ?? 1} ${lineCount === 1 || !lineCount ? 'line' : 'lines'}`}
+        </EuiText>,
+        <EuiText size="xs" color="subdued" className="queryEditor__footerItem">
+          {props.query.dataset?.timeFieldName || ''}
+        </EuiText>,
+        <QueryResult queryStatus={props.queryStatus!} />,
+      ],
+      end: [
+        <EuiButtonEmpty
+          iconSide="left"
+          iconType="clock"
+          iconGap="s"
+          size="xs"
+          onClick={toggleRecentQueries}
+          className="queryEditor__footerItem"
+          flush="both"
+        >
+          <EuiText size="xs" color="subdued">
+            {'Recent queries'}
+          </EuiText>
+        </EuiButtonEmpty>,
+      ],
+    },
+  };
 
-    const languageEditorFunc = this.languageManager.getLanguage(this.props.query.language)!.editor;
+  const languageEditorFunc = languageManager.getLanguage(props.query.language)!.editor;
 
-    const languageEditor = useQueryEditor
-      ? languageEditorFunc(singleLineInputProps, {}, defaultInputProps)
-      : languageEditorFunc(singleLineInputProps, singleLineInputProps, {
-          filterBar: this.props.filterBar,
-        });
+  const languageEditor = useQueryEditor
+    ? languageEditorFunc(singleLineInputProps, {}, defaultInputProps)
+    : languageEditorFunc(singleLineInputProps, singleLineInputProps, {
+        filterBar: props.filterBar,
+      });
 
-    return (
+  return (
+    <div
+      className={classNames(
+        props.className,
+        'osdQueryEditor',
+        isCollapsed ? 'collapsed' : 'expanded',
+        !languageEditor.TopBar.Expanded && 'emptyExpanded'
+      )}
+    >
+      <div
+        ref={bannerRef}
+        className={classNames('osdQueryEditor__banner', props.bannerClassName)}
+      />
       <div
         className={classNames(
-          className,
+          props.className,
           'osdQueryEditor',
-          this.state.isCollapsed ? 'collapsed' : 'expanded',
+          isCollapsed ? 'collapsed' : 'expanded',
           !languageEditor.TopBar.Expanded && 'emptyExpanded'
         )}
       >
-        <div
```

Comment:
Why did we move the location of the banner?

Reply:
whoops seems to have duplicated